anonlinkclient/cli.py

```diff
@@ -331,36 +336,75 @@ def create(name, project, apikey, output, threshold, server, retry_multiplier, r
-@click.argument('clk_json', type=click.File('rb'))
+@click.argument('clk_json', type=click.Path(exists=True, dir_okay=False))
 @click.option('--project', help='Project identifier')
 @click.option('--apikey', help='Authentication API key for the server.')
 @click.option('-o', '--output', type=click.File('w'), default='-')
 @click.option('--blocks', help='Generated blocks JSON file', type=click.File('rb'))
 @add_options(rest_client_option)
+@click.option('--profile', help="AWS profile to use if uploading to own S3 Bucket", default=None)
 @verbose_option
-def upload(clk_json, project, apikey, output, blocks, server, retry_multiplier, retry_max_exp, retry_stop, verbose):
-    """Upload CLK data to entity matching server.
+def upload(clk_json, project, apikey, output, blocks, server, retry_multiplier, retry_max_exp, retry_stop, profile, verbose):
+    """Upload CLK data to the Anonlink Entity server.
 
     Given a json file containing hashed clk data as CLK_JSON, upload to
     the entity resolution service.
 
-    Use "-" to read from stdin.
+    The following environment variables can be used to override default behaviour:
+
+    * UPLOAD_OBJECT_STORE_SERVER
+
     """
     msg = 'CLK and Blocks' if blocks else 'CLK'
+
     if verbose:
-        log("Uploading CLK data from {}".format(clk_json.name))
+        log("Uploading CLK data from {}".format(clk_json))
         log("Project ID: {}".format(project))
         log("Uploading {} data to the server".format(msg))
 
     rest_client = create_rest_client(server, retry_multiplier, retry_max_exp, retry_stop, verbose)
 
+    if verbose: log("Fetching temporary credentials")
+    res = rest_client.get_temporary_objectstore_credentials(project, apikey)
+    credentials = res['credentials']
+    upload_info = res['upload']
+
+    endpoint = os.getenv('UPLOAD_OBJECT_STORE_SERVER', upload_info['endpoint'])
+
+    object_store_credential_providers = []
+    if profile is not None:
+        object_store_credential_providers.append(FileAWSCredentials(profile=profile))
+
+    object_store_credential_providers.append(
+        Static(access_key=credentials['AccessKeyId'],
+               secret_key=credentials['SecretAccessKey'],
+               token=credentials['SessionToken']))
+
+    mc = Minio(
+        endpoint,
+        credentials=Credentials(provider=Chain(object_store_credential_providers)),
+        region='us-east-1',
+        secure=upload_info['secure']
+    )
+
+    if verbose:
+        log('Checking we have permission to upload')
+    mc.put_object(upload_info['bucket'], upload_info['path'] + "/upload-test", io.BytesIO(b"something"), length=9)
+
     # combine clk and blocks if blocks is provided
     if blocks:
-        out = combine_clks_blocks(clk_json, blocks)
+        with open(clk_json, 'rb') as encodings:
+            out = combine_clks_blocks(encodings, blocks)
         response = rest_client.project_upload_clks(project, apikey, out)
```

Comment:
Why we don't upload to minio as well like the no blocking case?

Reply:
In the initial [plan](https://github.com/data61/anonlink-entity-service/blob/develop/docs/designs/anonlink-upload-data-plan.md#modification-to-the-upload-endpoint) we will only support binary encodings being uploaded to the object store. Eventually a binary format for the blocks will also be implemented.

Reply:
Okay